File: Leetcode/dfs_recursion_backtracking/119_Pascal's TriangleII/Solution.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        LinkedList<Integer> last = new LinkedList<>();
        LinkedList<Integer> res = new LinkedList<>();
        res.add(1);
        for(int i = 1; i<=rowIndex; ++i){
            last = res;
            res = new LinkedList<Integer>();
            res.add(1);
            for(int j = 0; j<last.size()-1; ++j){
                res.add(last.get(j)+last.get(j+1));
            }
            res.add(1);
        }
        return res;
    }
}

File: Leetcode/dfs_recursion_backtracking/119_Pascal's TriangleII/Solution.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        last, res = [], [1]
        for i in range(1, rowIndex+1):
            last = res
            res = [1]+[last[i]+last[i+1] for i in range(len(last)-1)]+[1]
        return res
